mne/io/nirx/nirx.py

```diff
@@ -0,0 +1,204 @@
+        assert_one(file_mat, "Should be one mat file")
+
+        # Read number of rows of wavelength data which corresponds to
+        # number of samples
+        last_sample = -2
+        for line in open(file_wl1[0]):
+            last_sample += 1
+
+        # Read participant information file
+
+        inf = cp.ConfigParser(allow_no_value=True)
+        inf.read(file_inf)
+        inf = inf._sections['Subject Demographics']
+
+        # mne requires specific fields in here
+        # https://github.com/mne-tools/mne-python/ ...
+        # blob/master/mne/io/meas_info.py#L430
+        # TODO: Can you put more values in subject_info than specified in link?
+        #       NIRX also records "Study Type", "Experiment History",
+        #       "Additional Notes", "Contact Information"
+        subject_info = {}
+        subject_info['last_name'] = inf['name'].split()[-1].replace("\"", "")
+        subject_info['first_name'] = inf['name'].split()[0].replace("\"", "")
+        subject_info['middle_name'] = inf['name'].split()[-2].replace("\"", "")
+        subject_info['birthday'] = inf['age']
+        subject_info['sex'] = inf['gender'].replace("\"", "")
+        # Recode values
+        if subject_info['sex'] in {'M', 'Male', '1'}:
+            subject_info['sex'] = 1
+        if subject_info['sex'] in {'F', 'Female', '2'}:
+            subject_info['sex'] = 2
+        # NIRStar does not record an id, or handedness by default
+
+        # Read header file
+        # This is a bit tricky as the header file isnt compliant with
+        # the config specifications. So we need to remove all text
+        # between comments before passing to config parser
+        hdr_str = open(file_hdr[0]).read()
+        hdr_str = re.sub('#.*?#', '', hdr_str, flags=re.DOTALL)
+        hdr = cp.RawConfigParser()
+        hdr.read_string(hdr_str)
+
+        # Check that the file format version is supported
+        assert (hdr['GeneralInfo']['NIRStar'] == "\"15.2\""), \
+            "Only NIRStar version 15.2 is supported"
+
+        # Parse required header fields
+
+        # Extract source-detectors requested by user
+        sources = [int(s) for s in re.findall(r'(\d)-\d:\d+',
+                   hdr['DataStructure']['S-D-Key'])]
+        detectors = [int(s) for s in re.findall(r'\d-(\d):\d+',
+                     hdr['DataStructure']['S-D-Key'])]
+        sdindex = [int(s) for s in re.findall(r'\d-\d:(\d+)',
+                   hdr['DataStructure']['S-D-Key'])]
+        assert (len(sources) == len(detectors)), \
+            "Same amount of sources and detectors required"
+        assert (len(sources) == len(sdindex)), \
+            "Same amount of sources and keys required"
+
+        # Create mne structure
+        # TODO: ch_type is currently misc as I could not find appropriate
+        #       other type, the hbo and hbr type are not relevant until the
+        #       signal has been converted
```

Comment:
1. what are the units? Candela? If not, is it in this list?

   https://github.com/mne-tools/mne-python/blob/master/mne/io/constants.py#L740

   If not, we can add whatever the units actually are.

2. You can use the type as FNIRS:

   https://github.com/mne-tools/mne-python/blob/master/mne/io/constants.py#L170

   You get this automatically (I think) if you set `ch_types='fnirs'` below

3. We'll need to add a different "coil type" for these intensity measurements because, as you say, [HbO/HbR](https://github.com/mne-tools/mne-python/blob/master/mne/io/constants.py#L803) are not appropriate. So what would this "coil type" be? It's not an electrode, it's not an MEG SQUID or OPM -- what would you say is the physical thing / the physical things doing the measumerments?

Reply:
1. I am just checking this answer with the manafacturer to make sure I get the units correct (next step is to convert to optical density which is unitless)  
2. if I set the type to fnirs i get the error 
    > KeyError: "kind must be one of ['eeg', 'mag', 'grad', 'ref_meg', 'misc', 'stim', 'eog', 'ecg', 'emg', 'seeg', 'bio', 'ecog', 'hbo', 'hbr'], not fnirs"
3. We call them `optodes` :bulb:

Reply:
@larsoner in commit 4d922266106ca0f8abcd631beb0e3450a27cf118 I have taken an initial crack at adding two new types `fnirs_raw` (for the voltage measurement directly from machine) and `fnirs_od` (the middle step in conversion to chromophore). I also added an optode coil type. Could you please take a review of how I added these types and let me know if I am on the right track.